lib/einhorn.rb

```diff
@@ -236,6 +236,20 @@ def self.preload
         else
           log_info("Requiring #{path} (if this hangs, make sure your code can be properly loaded as a library)", :upgrade)
           require path
+
+          if Einhorn::State.config[:gc_for_cow_before_fork]
+            log_info("Starting GC to improve copy-on-write memory sharing")
+
+            # Make the GC more copy-on-write friendly by forcibly incrementing the generation
+            # counter on all objects to its maximum value. Learn more at: https://github.com/ko1/nakayoshi_fork
+            GC.start
+            3.times do
```

Comment:
Just for my own edification: it looks like we run the GC three times in this loop so what's the `GC.start` above it doing? Does it actually need four runs?

Reply:
Not sure, I'm mostly carrying over what we have internally that works. The gem itself does 4, so I assume the goal is to get rid of any short lived objects by doing a full sweep to start, and then do 3 more to ensure everything else is moved over?